subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/DefaultTaskClassInfoStore.java

```diff
@@ -150,6 +151,11 @@ public ClassLoader getClassLoader() {
         public String getActionClassName() {
             return type.getName();
         }
+
+        @Override
+        public String getDisplayName() {
+            return "Execute " + method.getName();
```

Comment:
The type is usually already in the task name/path. The task path is the parent in a build operation tree (see IDEA screenshot) and is appended for the full display name [here](https://github.com/gradle/gradle/pull/2618/files#diff-b58695f258ac9212686bd1998667290eR112).